### PR TITLE
Split eBPF programs

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
This PR is bringing a new eBPF release for netdata and it is the base for improvements we are bringing to process monitoring and file access.

This PR is also bringing an improvement for user navigation.
##### Component Name
ebpf.plugin
##### Test Plan

1 - Compile this PR
2 - Take a look on charts and links added to dashboard.

##### Additional Information
This PR is not bringing nothing really new, the biggest difference on kernel ring it is the fact we are loading two eBPF programs instead one, but the code tested here were previously tested.

### CODE REVIEW

**Review by hugovalente-pm — DISMISSED** [review]

**Review by UmanShahzad — APPROVED** [review]
👍 

Graphs & links work:

image

image

File: collectors/ebpf.plugin/ebpf_fd.c [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+                      NETDATA_EBPF_CHART_TYPE_LINE,
+                      NETDATA_CHART_PRIO_EBPF_FD_CHARTS,
+                      ebpf_create_global_dimension,
+                      fd_publish_aggregated,
+                      NETDATA_FD_SYSCALL_END,
+                      NETDATA_EBPF_MODULE_NAME_FD);
+
+    if (em->mode < MODE_ENTRY) {
+        ebpf_create_chart(NETDATA_FILESYSTEM_FAMILY,
+                          NETDATA_FILE_OPEN_ERR_COUNT,
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_fd.c:453]: I don't see any charts with open/close errors neither aggregated nor in the apps section.

File: collectors/ebpf.plugin/README.md [hunk 2]
```diff
@@ -210,20 +210,26 @@ When the integration is enabled, eBPF collector allocates memory for each proces
...
     files are not found.
 -   `disk` : This eBPF program creates charts that show information about disk latency independent of filesystem.
--   `filesystem`: This eBPF program creates charts that show latency information for selected filesystem.
--   `process`: This eBPF program creates charts that show information about process creation, calls to open files.
-    When in `return` mode, it also creates charts showing errors when these operations are executed.
--   `network viewer`: This eBPF program creates charts with information about `TCP` and `UDP` functions, including the
-    bandwidth consumed by each.
--   `sync`: Montitor calls for syscalls sync(2), fsync(2), fdatasync(2), syncfs(2), msync(2), and sync_file_range(2).
--   `vfs`: This eBPF program creates charts that show information about VFS (Virtual File System) functions.
+-   `filesystem` : This eBPF program creates charts that show information about some filesystem latency.
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply collectors/ebpf.plugin/README.md:230]: Thank you, I removed one.

File: collectors/ebpf.plugin/ebpf_fd.c (hunk 1 above)
Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf_fd.c:453]: I fixed this with last commit.

File: collectors/ebpf.plugin/ebpf_fd.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+typedef struct netdata_fd_stat {
+    uint64_t pid_tgid;                     //Unique identifier
+    uint32_t pid;                          //process id
+
+    //Counter
+    uint32_t open_call;                    //open syscalls (open and openat)
+    uint32_t close_call;                   //Close syscall (close)
+
+    //Counter
```
[the remark this answers is not in the document]
Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf_fd.h]: I also fixed this.

File: libnetdata/ebpf/ebpf.c [hunk 4]
```diff
@@ -541,10 +541,12 @@ static netdata_run_mode_t ebpf_select_mode(char *mode)
     return MODE_ENTRY;
 }
 
-void ebpf_update_module_using_config(ebpf_module_t *modules)
+void ebpf_update_module_using_config(ebpf_module_t *modules, int source)
```
Comment (UmanShahzad) [inline libnetdata/ebpf/ebpf.c:544]: I suppose it doesn't matter much for this case because the function is only used in 1 place, but I wonder if it should be documented what `source` is expected to be, e.g. `0 == from stock, 1 == from user`, etc.

**Review by UmanShahzad — DISMISSED** [review]

File: libnetdata/ebpf/ebpf.c (hunk 4 above)
Comment (thiagoftsm) [reply libnetdata/ebpf/ebpf.c:544]: We have some undocumented functions, I plan to bring a PR for this.

File: collectors/ebpf.plugin/ebpf_apps.h [hunk 5]
```diff
@@ -11,7 +11,7 @@
 #include "libnetdata/ebpf/ebpf.h"
 
 #define NETDATA_APPS_FAMILY "apps"
-#define NETDATA_APPS_FILE_GROUP "file (eBPF)"
+#define NETDATA_APPS_FILE_GROUP "file_access"
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_apps.h:14]: Why did you change that? You use `(eBPF)` everywhere.

File: collectors/ebpf.plugin/ebpf.c [hunk 6]
```diff
@@ -129,7 +129,7 @@ ebpf_module_t ebpf_modules[] = {
       .update_time = 1, .global_charts = 1, .apps_charts = CONFIG_BOOLEAN_NO, .mode = MODE_ENTRY,
       .optional = 0, .apps_routine = ebpf_fd_create_apps_charts, .maps = NULL,
       .pid_map_size = ND_EBPF_DEFAULT_PID_SIZE, .names = NULL, .cfg = &fd_config,
-      .config_file = NETDATA_SYNC_CONFIG_FILE},
+      .config_file = NETDATA_FD_CONFIG_FILE},
```
Comment (UmanShahzad) [inline collectors/ebpf.plugin/ebpf.c:132]: Just an FYI that I noticed while doing hard IRQs that many other entries in this table also use NETDATA_SYNC_CONFIG_FILE incorrectly. I'll fix that in my PR as a separate commit or make a new PR.

File: collectors/ebpf.plugin/ebpf_apps.h (hunk 5 above)
Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf_apps.h:14]: When we were using `file (eBPF)` the new links created to move between global charts and apps charts were not working, so we are renaming it to have conditions to create the needed links.

File: collectors/ebpf.plugin/ebpf.c (hunk 6 above)
Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf.c:132]: Than you, I fixed with last commit.

**Review by vlvkobal — APPROVED** [review]

**Review by UmanShahzad — APPROVED** [review]
